shiny/_main.py

```diff
@@ -261,6 +282,8 @@ def run_app(
 
     if reload_dirs is None:
         reload_dirs = []
+        if app_dir is not None:
+            reload_dirs = [app_dir]
```

Comment:
We should do a check below where we are calling `reload_dirs.append(app_dir)` to make sure `app_dir` is not already added. 

I can't suggest in this comment, so I'll put it here as code.

```python
        if app_dir and app_dir not in reload_dirs:
            reload_dirs.append(app_dir)
```